components/remote_settings/src/signatures.rs

```diff
@@ -0,0 +1,107 @@
+use core::clone::Clone;
+
+use crate::{RemoteSettingsRecord, Result};
+use rc_crypto::contentsignature;
+use serde_json::{json, Value};
+
+/// Remove `deleted` and `attachment` fields if they are null.
+fn select_record_fields(value: &Value) -> Value {
+    if let Value::Object(map) = value {
+        let new_map = map
+            .iter()
+            .filter_map(|(key, v)| {
+                if key == "deleted" || key == "attachment" && v.is_null() {
+                    None
+                } else {
+                    Some((key.clone(), v.clone()))
+                }
+            })
```

Comment:
Maybe splitting this out into a call to `filter()` and a call to `cloned()` would be simpler, but however you want to spell this out is fine with me.

Reply:
Could not use `clone()` because the compiler wouldn't let me clone the values of type `(&String, &Value)`. I left `.map(|(key, v)| (key.clone(), v.clone()))`